cpp/SnowflakeAzureClient.cpp

```diff
@@ -38,11 +38,13 @@ SnowflakeAzureClient::SnowflakeAzureClient(StageInfo *stageInfo,
   m_parallel(std::min(parallel, std::thread::hardware_concurrency()))
 {
   const std::string azuresaskey("AZURE_SAS_KEY");
-  char caBundleFile[MAX_PATH] ={0};
+  char caBundleFile[MAX_PATH] = {0};
   if(transferConfig && transferConfig->caBundleFile) {
-      int len = std::min((int)strlen(transferConfig->caBundleFile), MAX_PATH - 1);
-      sb_strncpy(caBundleFile, sizeof(caBundleFile), transferConfig->caBundleFile, len);
```

Comment:
sb_strncpy internally uses strncpy and g++82 compiler toolchain complains about it and bails out because of the compilation parameters used.

Reply:
Please use sb_strcpy then, it equals to strcpy on Linux so should be good to use.

Reply:
@SimbaGithub the issue with sb_strcpy is it internally uses strncpy which causes compiler error when using g++82 
```
In function ‘char* sb_strncpy(char*, size_t, const char*, size_t)’,
    inlined from ‘Snowflake::Client::SnowflakeAzureClient::SnowflakeAzureClient(Snowflake::Client::StageInfo*, unsigned int, size_t, Snowflake::Client::TransferConfig*)’ at /home/vreddy/libsnowflakeclient/cpp/SnowflakeAzureClient.cpp:44:17:
/home/vreddy/libsnowflakeclient/include/snowflake/Simba_CRTFunctionSafe.h:227:23: error: ‘char* strncpy(char*, const char*, size_t)’ specified bound depends on the length of the source argument [-Werror=stringop-overflow=]
         return strncpy(out_dest, in_src, in_sizeToCopy);
```

Reply:
That's sb_strncpy, not sb_strcpy. sb_strcpy should work.

Reply:
@sfc-gh-vreddy Please use sb_strcpy, not sb_strncpy.

Reply:
Got it. Using sb_strcpy and validating the return value too.